### PR TITLE
Update controller-{runtime, tools} / Kubernetes 1.12

### PR DESCRIPTION
**Pull request by vincepri**
**What this PR does / why we need it**:
This PR updates the controller-runtime and controller-tools vendored dependencies to their latest version, updates the Go version to 1.11.2 and Kubernetes to 1.12. The updated tools also remove the need for some hacks in the Makefile.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by vincepri** [comment]
Seems this is blocked by the CI image being outdated. Investigating.

**Comment by vincepri** [comment]
It seems that the image used for `pull-cluster-api-make` uses Go 1.9.3.

```
$ docker run -it --entrypoint bash gcr.io/k8s-testimages/kubekins-e2e:v20181205-915278e90-1.10
root@3fe4df991aa6:/workspace# go version
go version go1.9.3 linux/amd64
```

File: Gopkg.toml [hunk 1]
```diff
@@ -24,11 +24,11 @@ required = [
 
 [[constraint]]
   name="sigs.k8s.io/controller-runtime"
-  version="v0.1.1"
+  version="v0.1.8"
```
Comment (alvaroaleman) [inline Gopkg.toml]: This is not needed, the current constraint does not have an operator which means `>=v0.1.1,<2.0.0`[0]

We are already at `v0.1.6` for controller-tools and `v0.1.7` for controller-runtime

[0]https://golang.github.io/dep/docs/Gopkg.toml.html

Comment (vincepri) [reply Gopkg.toml]: That's a fair point. Given the breaking changes though I preferred to modify the versions and update afterwards. Any thoughts around that? I'm happy to change if needed.

Comment (alvaroaleman) [reply Gopkg.toml]: Well the line above reads `# STANZAS BELOW ARE GENERATED AND MAY BE WRITTEN - DO NOT MODIFY BELOW THIS LINE.` so IMHO we should just leave it as-is

**Comment by justinsb** [comment]
Let's do it!  I think we need to figure out what we do about regenerating the boilerplate, but maybe we just don't have to.  I'm thinking in particular that later versions of kubebuilder set up a metrics endpoint, and I don't think we have it because we have the "old" scaffolding.  But maybe that's a good thing as we probably shouldn't be automatically opening up more ports without a conscious effort.

/approve
/lgtm

**Pull request merged**